src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -114,5 +137,50 @@ private bool ShouldRestoreOnBuild
                 return packageRestoreConsent.IsAutomatic;
             }
         }
+
+        #region IVsUpdateSolutionEvents4
+
+        public void UpdateSolution_QueryDelayFirstUpdateAction(out int pfDelay)
+        {
+            // check if NuGet lock is already acquired by some other NuGet operation
+            if (LockService.IsLockHeld)
+            {
+                // delay build by setting pfDelay to non-zero
+                pfDelay = 1;
+            }
+            else
+            {
+                var isLockAcquired = LockService.TryAcquireLock(NuGetLockTimeout, out _nuGetLock);
+
+                if (isLockAcquired)
+                {
+                    // Set delay to 0 which means don't delay build
+                    pfDelay = 0;
+                }
+                else
+                {
+                    // delay build since NuGet lock is still held by other operation
+                    pfDelay = 1;
+                }
+            }
+        }
+
+        public void UpdateSolution_BeginFirstUpdateAction() { }
+
+        public void UpdateSolution_EndLastUpdateAction()
+        {
+            // Release NuGet lock so that other NuGet operation can continue now.
+            _nuGetLock?.Dispose();
```

Comment:
Again how would that make a difference? Anyways it can't be used again after disposed besides it will always come here when it has actually acquired the lock and set `_nugetLock `to new IDisposible instance.